cdap-proto/src/main/java/co/cask/cdap/proto/metadata/lineage/FieldOperationInput.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.metadata.lineage;
+
+import co.cask.cdap.api.lineage.field.EndPoint;
+import co.cask.cdap.api.lineage.field.InputField;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+/**
+ * Represents input to the field operation.
+ */
+public class FieldOperationInput {
+  private final List<EndPoint> endPoints;
+  private final List<InputField> fields;
+
+  public FieldOperationInput(@Nullable List<EndPoint> endPoints, @Nullable List<InputField> fields) {
```

Comment:
isn't it either a list of endpoints or a list of fields? So you can have two constructors, one for each case. That will enforce that it is either/or.

Reply:
I tried with the two constructors, however since both are of type List and java does type erasure, compilation fails.